lib/qm-dsp/dsp/keydetection/GetKeyMode.cpp

```diff
@@ -266,8 +266,10 @@ int GetKeyMode::process(double *PCMData)
   std::cout << std::endl;
 */
     double dummy;
-    // '1 +' because we number keys 1-24, not 0-23.
-    key = 1 + (int)ceil( (double)MathUtilities::getMax( m_Keys, 2* m_BPO, &dummy )/3 );
+    // m_Keys[1] is C center  1 / 3 + 1 = 1
+    // m_Keys[4] is D center  4 / 3 + 1 = 2
+    // '+ 1' because we number keys 1-24, not 0-23.
+    key = MathUtilities::getMax( m_Keys, 2* m_BPO, &dummy ) / 3 + 1;

```

Comment:
Now the conversion from `double` to `int` is implicit and instead of a doing it by applying `ceil()` you now implicitly apply `floor()` (assuming the value is always >= 0). I suggest to make this more calculation more transparent.

Reply:
Get Max already returns an integer.
The number of the maximum bin, which can be the full half note, or 1/3 offset in both directions.

If we assume that Is a western track played in tune we should never get results from these 1/3 bins.
I am wondering if we doing it right to assume a full key even if the maximum is off. This is the same as if we pitch the track by 2%. 

Alternative we can valuate the side bins only by 50 percent and also return fractional keys. I am curious how many tracks would be effected.

An other issue could be if let's say c+1/3 is the max, but full d is higher than full c. Currently the heavily out of tune c is returned but could it be a d track?

We may considder to make this configurable.

Reply:
A configuration option would overwhelm  a casual user like me. Let's stay with a sensible default.